compiler/src/main/kotlin/walkers/symbol.kt

```diff
@@ -3,14 +3,29 @@ package dk.aau.cs.d409f19.cellumata.walkers
 import dk.aau.cs.d409f19.cellumata.ast.*
 import java.lang.Exception
 
+/**
+ * Thrown when a undefined symbol is encountered. This exception indicates there is a use-before-declaration scenario.
+ */
 class SymbolException(val ident: String) : Exception("\"$ident\" was used before it was declared")
 
-class ScopeCheckVisitor(val symbolTable: SymbolTable) : BaseASTVisitor() {
+/**
+ * Walks through the abstract syntax tree, extracts symbols, and checks for use-before-declaration.
```

Comment:
Missing explanation of what is meant by `symbolTableSession`, namely, what is meant by "session"?

Reply:
> symbolTableSession is the symbol insertions session that keeps track of where to create scopes and insert symbols

How about this?

Reply:
On second thought, the existing is fine.